keyboards/yandrstudio/buff67v3/buff67v3.c

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2022 JasonRen(biu)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "buff67v3.h"
+
+
+#ifdef RGBLIGHT_ENABLE
+
+const rgblight_segment_t PROGMEM my_capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_RED}
+);
+
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    my_capslock_layer
+);
+
+bool led_update_kb(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    return true;
+}
+
+void keyboard_post_init_kb(void) {
+    rgblight_reload_from_eeprom();
+    rgblight_layers = my_rgb_layers;
+}
+
+#endif
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        case LOCK_GUI:
+            process_magic(GUI_TOG, record);
+            return false;
```

Comment:
Why?  

Why not just use GUI_TOG.

Reply:
To be blunt, "how it shows up on via" isn't a good reason for this implementation, ESPECIALLY at the keyboard level.   You are potentially breaking expected behavior for anyone that uses the board, and wants to use F23.  

If you absolutely must implement this, then you need to move this implementation to the via keymap. And there you can use USER00-USER15 to use a custom keycode in the via json.

Reply:
thanks，i have used USER00. It's really troublesome that via's documentation didn't mention this.